pkg/gcp/plugin.go

```diff
@@ -493,13 +496,33 @@ func (s *GCPPluginServer) _GetUsedAddressSpaces(ctx context.Context, req *paragl
 				resp.AddressSpaceMappings[i].AddressSpaces = append(resp.AddressSpaceMappings[i].AddressSpaces, *secondaryRange.IpCidrRange)
 			}
 		}
+
+		// Get addresses not associated with the vpc (might be used for PSCs)
+		listAddressesReq := &computepb.ListAddressesRequest{
+			Project: project,
+			Filter:  proto.String(fmt.Sprintf("labels.paraglider_ns eq %s", deployment.Namespace)), // TODO NOW: make the label name a constant
+		}
+		listAddressesResp := addressesClient.List(ctx, listAddressesReq)
```

Comment:
Is this query a superset of the for loop above where we query for the exact VPC/subnets for each deployment? If so, could we get rid of the above for loop? If we do both then I assume it would actually create duplicates?

Reply:
Nope, these are separate resource types. In this one, I am searching for Addresses (IP adresses) vs. searching for VPCs and getting their address spaces. VPC address spaces are not Addresses (weirdly :) )